app/src/main/java/org/sagebionetworks/research/mpower/room/RoomTypeConverters.kt

```diff
@@ -0,0 +1,106 @@
+package org.sagebionetworks.research.mpower.room
+
+import android.arch.persistence.room.TypeConverter
+
+import com.google.gson.reflect.TypeToken
+import org.joda.time.DateTime
+import org.sagebionetworks.bridge.rest.RestUtils
+import org.sagebionetworks.bridge.rest.model.ActivityType
+import org.sagebionetworks.bridge.rest.model.ScheduleStatus
+import org.sagebionetworks.bridge.rest.model.ScheduledActivityListV4
+
+/**
+ * This class controls how objects are converted to and from data types supported by SqlLite
+ * Room recognizes which class types can be converted by
+ * the @TypeConverter annotation, and inferred by the method structure
+ */
+class RoomTypeConverters {
+
+    private val schemaRefListType = object : TypeToken<List<RoomSchemaReference>>() {}.type
+    private val surveyRefListType = object : TypeToken<List<RoomSurveyReference>>() {}.type
+
+    @TypeConverter
+    fun fromTimestamp(value: Long?): DateTime? {
+        val valueChecked = value ?: return null
+        return DateTime(valueChecked)
+    }
```

Comment:
If we convert this to/from long, we lose timezone information. Is that something that we care about? If it is, should we consider converting to/from an ISO string instead?

Reply:
Good point. 

We also have some tricky logic when Bridge sends a DateTime but really means LocalDateTime or LocalDate, not obvious how to query that. Hopefully we don't have to do a rough approximate query and perform additional filtering